src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -230,8 +229,9 @@ public Map<String, Object> getProcessedRecordForStreamingIngest(SinkRecord recor
         if (metadataConfig.allFlag) {
           streamingIngestRow.put(TABLE_COLUMN_METADATA, MAPPER.writeValueAsString(row.metadata));
         }
-      } catch (IOException e) {
+      } catch (JsonProcessingException e) {
         // return an exception and propagate upwards
+        // TODO tzhang: do we need to rethrow?
```

Comment:
@sfc-gh-japatel I believe we need to rethrow the exception here, don't we?

Reply:
It can only throw exception in case `MAPPER.writeValueAsString` throws exception. 
Is there a need to change this from IoException to JsonProcessExc...?

Reply:
We will need to catch the exception and put it into DLQ? Should this part of separate change?

Reply:
hmmm, this is your code, not from Zekai, it used to be JsonProcessingException, and he changed it to IoException which i feel like is by accident. And we didn't thrown so I'm not sure what you mean by "put it into DLQ", could you elaborate?

Reply:
Catching JsonExcetion makes sense. Now that I am looking and reading it again,seems like just swallowing it will continue the connector.. 

When I said, put it into DLQ, I meant rather than just continuing with it, does it makes sense to rethrow it and make it deliverable to DLQ. 

Doesnt necessarily need to be in this change as I am not blocking here, just trying to understand how this will work. 

Please file a JIRA.. Now that we will have a schematization go throw same logic, we will need to carefully process this. Earlier the chances of it throwing an exception were less since we were just converting everything into JSON.

Reply:
> When I said, put it into DLQ, I meant rather than just continuing with it, does it makes sense to rethrow it and make it deliverable to DLQ?

Yes, I think we should, and that's the reason I pinged you here. Looks like now we may just skipping rows which is not correct, I will update it in this PR.

Reply:
Thanks Toby!

Reply:
This is not easy, will do in the next PR